euphonic/readers/castep.py

```diff
@@ -671,13 +671,12 @@ def record_mark_read(file_obj):
                 data = data.astype(np.int32)
             elif 'f' in dtype:
                 data = data.astype(np.float64)
+        elif 'i' in dtype:
```

Comment:
Another `match`?

Reply:
Not so clear-cut here as we are looking at more than one piece of information.

Maybe this whole bit can be structured better, but out of scope here if so.